backend/tests/test_course_build_processor.py

```diff
@@ -0,0 +1,478 @@
+            "receive_message",
+            {
+                "Messages": [
+                    {
+                        "ReceiptHandle": "message1",
+                        "Body": json.dumps(mock_sqs_data),
+                    }
+                ]
+            },
+            expected_params={
+                "QueueUrl": "https://testqueue",
+                "MaxNumberOfMessages": 10,
+                "WaitTimeSeconds": 20,
+            },
+        )
+        sqs_stubber.add_response(
+            "delete_message",
+            {},
+            expected_params={
+                "QueueUrl": "https://testqueue",
+                "ReceiptHandle": "message1",
+            },
+        )
+
+        sqs_stubber.activate()
+        mocker_map = {"sqs": sqs_client}
+        mocker.patch("boto3.client", lambda client: mocker_map[client])
+        mocker.patch("sys.argv", [""])
+        course_build_processor.main()
+
+    assert exc_info.type == Exception
+
+
+def test_course_build_status_processing(mocker, db, create_course_builds):
+    course_builds = db.query(CourseBuild).all()
+
+    course_build = course_builds[1]
+
+    with pytest.raises(
+        Exception,
+        match=f"Course build id: {course_build.id} status is processing",
+    ) as exc_info:
+        sqs_client = boto3.client("sqs", region_name="azeroth")
+        sqs_stubber = botocore.stub.Stubber(sqs_client)
+
+        mock_sqs_data = {"course_build_id": course_build.id}
+
+        mock_settings = mocker.Mock()
+        setattr(mock_settings, "SQS_QUEUE", "testqueue")
+        setattr(mock_settings, "SQS_POLL_INTERVAL_MINS", "1")
+        mocker.patch(
+            "rope.api.processors.course_build_processor.settings",
+            mock_settings,
+        )
+
+        engine = create_engine("postgresql://pguser:pgpassword@localhost/ropedb")
+        mock_factory = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+        mocker.patch(
+            "rope.api.processors.course_build_processor.get_db",
+            lambda: mock_factory,
+        )
+
+        sqs_stubber.add_response(
+            "get_queue_url",
+            {"QueueUrl": "https://testqueue"},
+            expected_params={"QueueName": "testqueue"},
+        )
+        sqs_stubber.add_response(
+            "receive_message",
+            {
+                "Messages": [
+                    {
+                        "ReceiptHandle": "message1",
+                        "Body": json.dumps(mock_sqs_data),
+                    }
+                ]
+            },
+            expected_params={
+                "QueueUrl": "https://testqueue",
+                "MaxNumberOfMessages": 10,
+                "WaitTimeSeconds": 20,
+            },
+        )
+        sqs_stubber.add_response(
+            "delete_message",
+            {},
+            expected_params={
+                "QueueUrl": "https://testqueue",
+                "ReceiptHandle": "message1",
+            },
+        )
```

Comment:
I think this expectation is actually incorrect but doesn't get flagged since right now you don't call `assert_no_pending_responses()`.